app/models/piece.rb

```diff
@@ -42,8 +42,15 @@ def blocked?(x, y)
     end
 
     piece_path.each do |square|
-      return true if game.is_occupied?(square[0], square[1])
+      return true if game.square_occupied?(square[0], square[1])
     end
     false
   end
+
+  def move_to!(x, y) # move and capture
+    @game = game
+    return false unless square_occupied?(x, y)
```

Comment:
Thanks for the review! Not redundant, the more explanation I get, the better. The original code looked similar to this but A: Rubocop kept complaining about using if-vs-unless/guard clauses, and so B: I second guessed myself and started researching and editing the code :)  Question about valid_move? Does that cover all of the pieces's moves that we are working on this week?
`def move_to!(x, y) # move and capture
    game
    return false if !valid_move?(x, y)
    if square_occupied?(x, y)
      target_piece = @game.piece.find_by(x_position: x, y_position: y)
      target_piece.update_attributes(x_position: nil, y_position: nil, captured: true)
    end
  end`